src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && herbivore.hidden === false) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const diedAnimal = Animal.alive.indexOf(herbivore);
+
+      Animal.alive.splice(diedAnimal, 1);
```

Comment:
[CODE PERFOMANCE] - Instead of using `indexOf` and `splice` to remove a dead animal, which involves a nested loop, it's recommended to use the `filter` method. This will improve performance, especially when dealing with a large number of animals.